### PR TITLE
[GOB-1027] Generate geometry for appartementsrechten

### PR DESCRIPTION
**Pull request by barykine**

### CODE REVIEW

File: src/data/sql/brk/kot.update_a_geometrie.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            AND kot1.expiration_date IS NULL
+            AND kot1.sectie = kot2.sectie
+            AND kot1.kad_gemeentecode = kot2.kad_gemeentecode
+        GROUP BY kot1.nrn_kot_id) q1),
+     vbo_kot_geometrie AS (
+        SELECT distinct ON(kot.nrn_kot_id)
+           kot.nrn_kot_id AS nrn_kot_id,
+           vbo.geometrie AS geometrie
+        FROM brk_prep.kadastraal_object kot
+        JOIN json_array_elements(kot.adressen::json) AS adres
```
Comment (JKoetsier) [inline src/data/sql/brk/kot.update_a_geometrie.sql]: Why not use jsonb_array_elements? That takes away the need to cast to json.

File: src/data/sql/brk/kot.update_a_geometrie.sql [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            AND kot.expiration_date IS NULL
+            AND vbo.geometrie IS NOT NULL
+            AND (g_poly IS NULL OR ST_Within(vbo.geometrie, g_poly) = true))
+UPDATE brk_prep.kadastraal_object
+SET geometrie = vbo_kot_geometrie.geometrie
+FROM vbo_kot_geometrie
+WHERE brk_prep.kadastraal_object.nrn_kot_id = vbo_kot_geometrie.nrn_kot_id
+    AND brk_prep.kadastraal_object.index_letter = 'A'
+    AND brk_prep.kadastraal_object.geometrie IS NULL
+    AND brk_prep.kadastraal_object.expiration_date IS NULL
```
Comment (JKoetsier) [inline src/data/sql/brk/kot.update_a_geometrie.sql]: The checks on index_letter and expiration_date are redundant; vbo_kot_geometrie already includes that check.

File: src/data/sql/brk/kot.update_a_geometrie.sql [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    WHERE kot1.index_letter = 'A'
+        AND kot1.expiration_date IS NULL
+        AND kot1.sectie = kot2.sectie
+        AND kot1.kad_gemeentecode = kot2.kad_gemeentecode
+    GROUP BY kot1.nrn_kot_id)
+UPDATE brk_prep.kadastraal_object
+SET geometrie = center_g_poly.geometrie
+FROM center_g_poly
+WHERE brk_prep.kadastraal_object.nrn_kot_id = center_g_poly.nrn_kot_id
+    AND brk_prep.kadastraal_object.index_letter = 'A'
```
Comment (JKoetsier) [inline src/data/sql/brk/kot.update_a_geometrie.sql]: Same here. index_letter and expiration_date are redundant.

File: src/data/sql/brk/kot.update_a_geometrie.sql [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+SET geometrie = vbo_kot_geometrie.geometrie
+FROM vbo_kot_geometrie
+WHERE brk_prep.kadastraal_object.nrn_kot_id = vbo_kot_geometrie.nrn_kot_id
+    AND brk_prep.kadastraal_object.index_letter = 'A'
+    AND brk_prep.kadastraal_object.geometrie IS NULL
+    AND brk_prep.kadastraal_object.expiration_date IS NULL
+;
+
+--  Then we update kadastrale objects of type 'A' without geometrie
+--  using ST_Centroid of related G-perceel.
```
Comment (JKoetsier) [inline src/data/sql/brk/kot.update_a_geometrie.sql]: related G-percelen

File: src/data/sql/brk/kot.update_a_geometrie.sql [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    WHERE kot1.geometrie IS NULL
+        AND kot1.index_letter = 'A'
+        AND kot1.expiration_date IS NULL)
+UPDATE brk_prep.kadastraal_object
+SET geometrie = near_a_poly.geometrie
+FROM near_a_poly
+WHERE brk_prep.kadastraal_object.nrn_kot_id = near_a_poly.nrn_kot_id
+    AND brk_prep.kadastraal_object.index_letter = 'A'
+    AND brk_prep.kadastraal_object.geometrie IS NULL
+    AND brk_prep.kadastraal_object.expiration_date IS NULL
```
Comment (JKoetsier) [inline src/data/sql/brk/kot.update_a_geometrie.sql]: index_ltter, geometrie and expiration_date are redundant.

**Comment by JKoetsier** [comment]
Nicely simplified :) Added some comments to improve understandability of the code.

File: src/data/sql/brk/kot.update_a_geometrie.sql (hunk 1 above)
Comment (barykine) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: done

File: src/data/sql/brk/kot.update_a_geometrie.sql (hunk 2 above)
Comment (barykine) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: done

File: src/data/sql/brk/kot.update_a_geometrie.sql (hunk 3 above)
Comment (barykine) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: done

File: src/data/sql/brk/kot.update_a_geometrie.sql (hunk 5 above)
Comment (barykine) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: done

File: src/data/sql/brk/kot.update_a_geometrie.sql (hunk 1 above)
Comment (JKoetsier) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: I think you still need to push the update ;)

Comment (barykine) [reply src/data/sql/brk/kot.update_a_geometrie.sql]: done

**Pull request merged**